### PR TITLE
Modal: Fix restore focus behavior

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
For whatever reason, the `restoreFocus` prop on react-aria's `FocusScope` component doesn't seem to work in this case, so I wrote a workaround.

### CODE REVIEW

**Review by ashharrison90 — COMMENTED** [review]
wonder if switching to use `OverlayContainer` instead of `Portal` would help? like https://github.com/grafana/grafana/pull/43746

**Comment by kaydelaney** [comment]
> wonder if switching to use `OverlayContainer` instead of `Portal` would help? like #43746

I'll give it a shot!

**Comment by kaydelaney** [comment]
Seems like that did a trick, though I did have to futz around with the backdrop's z-index a bit. Might be a little fragile, what do you reckon?

File: packages/grafana-ui/src/components/Modal/getModalStyles.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export const getModalStyles = stylesFactory((theme: GrafanaTheme2) => {
     `,
     modalBackdrop: css`
       position: fixed;
+      z-index: 1020;
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Modal/getModalStyles.ts:31]: i think it's fine, just need to get this from the theme instead. something like this?
```suggestion
      z-index: ${theme.zIndex.modalBackdrop};
```

**Review by ashharrison90 — APPROVED** [review]
lgtm, minor tweak so we're not hardcoding

File: packages/grafana-ui/src/components/Modal/getModalStyles.ts (hunk 1 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/components/Modal/getModalStyles.ts:31]: Ah nice, yeah, that'll do the trick

**Pull request merged**